### PR TITLE
Catch exceptions that occur when there is no network available

### PR DESCRIPTION
**Pull request by pimveldhuisen**
Fixes https://github.com/Tribler/tribler/issues/2225

### CODE REVIEW

File: util.py [hunk 1]
```diff
@@ -294,11 +294,15 @@ def get_lan_address_without_netifaces():
...
-    local_ip = s.getsockname()[0]
-    s.close()
-    return local_ip
+    try:
+        s = socket(AF_INET, SOCK_DGRAM)
+        s.connect(("192.0.2.0", 80)) # TEST-NET-1, guaranteed to not be connected => no callbacks
+        local_ip = s.getsockname()[0]
+        s.close()
+        return local_ip
+    except Exception as exception:
```
Comment (lfdversluis) [inline util.py]: General exception, should only catch what can be thrown. I believe it throws a socket.gaierror if not being able to connect.

**Comment by pimveldhuisen** [comment]
@lfdversluis Is this better?

**Comment by lfdversluis** [comment]
Yes 👍

**Comment by pimveldhuisen** [comment]
Cool, squashed it

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
HTTP ERROR 404

Problem accessing /job/dispersy/job/GH/job/PR/job/Allchannel+ChannelCommunity_short/12//artifact/output/quality_report.html. Reason:

```
Not Found
```

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by lfdversluis** [comment]
@pimveldhuisen The allchannel experiment fails in 4/5 runs because apparently only on 1 of the 5 clusters is R installed, the others will fail therefor. I think if pylint and the tests are happy it's ok (@whirm ?)

**Comment by pimveldhuisen** [comment]
Allchannel was not the only problem here, the first time the tests were aborted, the second time the pylint report resulted in a 404.

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by whirm** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by whirm** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
```
ERROR: [GitHub Commit Status Setter] - {"message":"No commit found for SHA: 52d68d5ed9ecac13ca8b120865f84737d0d1a876","documentation_url":"https://developer.github.com/v3/repos/statuses/"}, setting build result to FAILURE
Build step 'Set build status on GitHub commit [deprecated]' changed build result to FAILURE
```

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by whirm** [comment]
@pimveldhuisen I think this happens if you rebase while the tests are running?

**Comment by pimveldhuisen** [comment]
@whirm That actually seems to work fine (at least on Tribler PR's). I will avoid it here anyways though.

**Comment by whirm** [comment]
No need to avoid, the run resulting from pushing the rebase should finish afterwards with the actual results I guess.

**Comment by pimveldhuisen** [comment]
I get the same error:

```
Setting commit status on GitHub for https://github.com/Tribler/dispersy/commit/52d68d5ed9ecac13ca8b120865f84737d0d1a876
Setting commit status on GitHub for https://github.com/Tribler/tribler/commit/52d68d5ed9ecac13ca8b120865f84737d0d1a876
ERROR: [GitHub Commit Status Setter] Failed to update commit state on GitHub. Ignoring exception [{"message":"No commit found for SHA: 52d68d5ed9ecac13ca8b120865f84737d0d1a876","documentation_url":"https://developer.github.com/v3/repos/statuses/"}]
```

It seems it is looking for the dispersy commit in the tribler repo?  Maybe jenkins in misconfigured?

**Comment by whirm** [comment]
Nope. Apparently it takes the shotgun approach and just sends the updates to all checked out repositories (Tribler is checked out so the AllChannel experiment can be executed).

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
retest this please

**Pull request merged**